### DISCUSSION TITLE
Object assertion failing due to unknown reason (objects look the same)

### DISCUSSION THREAD

**Discussion by sjanisz** [fluentassertions/fluentassertions#2294] [Q&A]
Hello,

What could be the cause of failing assertion in case of following exception?

> Expected property actual.OriginalObject to be {
        "name": "job_unfiltered",
        "id": 0,
        "type": "field"
      }, but found {
        "name": "job_unfiltered",
        "id": 0,
        "type": "field"
      }.

Both objects look exactly the same

**Comment by jnyrup** [comment]
Just because their stringification looks the same doesn't mean the objects are considered equal.
Without any code we can only guess.

**Comment by sjanisz** [comment]
You are right, there was a custom method for deserializing string objects, as I just wanted to compare json objects I have just used JObject parsing:
`JObject.Parse(actual).Should().BeEquivalentTo(JObject.Parse(expected));`
instead of deserializing to actual objects

**Reply by dennisdoomen** [reply]
I've just tried this:

```cs
var actual = @"{ ""Property"": ""Hello"" }";

var expected = @"{ ""Property"": ""Hello"" }";

JObject.Parse(actual).Should().BeEquivalentTo(JObject.Parse(expected));
```

This succeeds.
